src/Server.php

```diff
@@ -197,6 +197,26 @@ protected function detectReturnType()
         }
     }
 
+    /**
+     * Initialize a user session and redirect back
+     */
+    public function initializeSession()
+    {
+        if (empty($_REQUEST['redirect'])) return $this->fail("No redirect specified", 400);
+        if (empty($_REQUEST['broker'])) return $this->fail("No broker specified", 400);
+
+        $checksum = $this->generateChecksum('initializeSession', $_REQUEST['broker'], $_REQUEST['redirect']);
+
+        if (empty($_REQUEST['checksum']) || $checksum != $_REQUEST['checksum']) {
+            return $this->fail("Invalid checksum", 400);
+        }
+
+        $this->startUserSession();
```

Comment:
Is the session cookie still there? Is the session id unchanged? If so, you're simply recreating the session file. Did it get deleted?
If you call this without a pre-existing session id, the new session isn't attached to the broker.

Reply:
Meta redirect causes an actual HTML render (visible page to the user) which for Safari is a reason to accept this as a page where cookies are allowed to be set on... That's the whole purpose of this PR.